.github/workflows/autograding.yml

```diff
@@ -20,54 +20,25 @@ jobs:
       uses: actions/setup-python@v5
       with:
         python-version-file: 'pyproject.toml' # Read python version from a file pyproject.toml
+        cache: 'pip'
```

Comment:
I think it should be `poetry` if we are using pyproject.toml, no?

Reply:
changing 'pip' with 'poetry' result in the following error message: `Error: Unable to locate executable file: poetry. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.`

Using the poetry action probable would solve this, but I leave it for a later re-work of the code.